tasks/administration/install/script.nix

```
@@ -0,0 +1,65 @@
+{ pkgs, ... }:
+
+# Task to INSTALL the specified derivation on current system including the firmware in a fully declarative way
+
+# Refer to https://github.com/nix-community/disko/issues/657#issuecomment-2146978563 for implementation notes
+
+{
+	perSystem = { system, inputs, ... }: {
+		install-task = pkgs.writeShellScriptBin "install-task" ''
+			system="$1"
+			disk="$2"
+
+			[ -n "$AGE" ] || AGE="${pkgs.age}/bin/age"
+			[ -n "$DISKO_INSTALL" ] || DISKO_INSTALL="${inputs.disko.packages.${system}.disko-install}/bin/disko-install"
```

If you change the parameter name to `inputs'` you get to use some un-system-ing logic:

```suggestion
			[ -n "$DISKO_INSTALL" ] || DISKO_INSTALL="${inputs'.disko.packages.disko-install}/bin/disko-install"
```